eng/testing/RunnerTemplate.sh

```diff
@@ -132,7 +132,13 @@ if [[ "$(uname -s)" == "Darwin" ]]; then
   # files already in /cores/ at this point. This is being done to prevent
   # inadvertently flooding the CI machines with dumps.
   if [[ ! -d "/cores" || ! "$(ls -A /cores)" ]]; then
-    ulimit -c unlimited
+    # Disabling core dumps on macOS. System dumps are large (even for very small
+    # programs) and not configurable. As a result, if a single PR build causes a
+    # lot of tests to crash, we can take out the entire queue.
+    # See discussions in:
+    #   https://github.com/dotnet/core-eng/issues/15333
+    #   https://github.com/dotnet/core-eng/issues/15597
+    ulimit -c 0
```

Comment:
This will disable system cores - are there any places where we hook up the runtimes coredump features like we do for the coreclr style tests? I am not aware of any, and they shouldn't be as bad in terms of size. Those could let us still get dumps reasonably.

Reply:
There's two things: we should remove this https://github.com/dotnet/runtime/blob/512a9ffcb8aa99ffc25fac76dc0c431543e45ba9/src/tests/Common/Coreclr.TestWrapper/CoreclrTestWrapperLib.cs#L223 and yeah, we are using `--with-heap`. That's pretty much grab everything in the managed process. It gives us the best results, but if the size is too large maybe we should consider dropping to `--normal` or `--triage` to at least get stacks.

Reply:
It would be great if the libraries tests could automatically wrap in coredump. I can't remember ever needing a dump from any process other than the test process (or possibly a child). So we could leave system dumps off at that point.

Reply:
Also update here and set `DbgMiniDumpType`, I expect:
https://github.com/dotnet/runtime/blob/03494f3dcaefc3dcdbb281a19db7633c750b921c/src/tests/Common/testenvironment.proj#L79-L81

Reply:
Filed https://github.com/dotnet/runtime/issues/65422